Legobot/Connectors/Discord.py

```diff
@@ -0,0 +1,318 @@
+        """
+        Sends heartbeat message to Discord
+
+        Attributes:
+            ws: Websocket connection to discord
+            seq: Sequence number of heartbeat
+        """
+
+        payload = {'op': 1, 'd': seq}
+        payload = json.dumps(payload)
+        logger.debug("Sending heartbeat with payload {}".format(payload))
+        ws.send(payload)
+        return
+
+    def update_sequence(self, seq):
+        """
+
+        Attributes:
+            seq (int): Sequence number of heartbeat
+        """
+
+        self.sequence = seq
+        logger.debug("Heartbeat sequence updated to {}".format(self.sequence))
+        return
+
+    def run(self):
+        while True:
+            self.send(self.ws, self.sequence)
+            time.sleep(self.interval)
+
+
+class DiscoBot(threading.Thread, object):
+    """
+    Relays messages between Discord and the Legobot baseplate.
+    Uses the REST API and Websockets. No plans to use async.
+
+    Attibutes:
+        actor_urn: URN (unique ID) of actor that initialized the bot
+        baseplate: Parent lego/actor
+        headers: HTTP headers use for calls to Discord REST endpoints
+        rest_baseurl: Base URL to the REST API
+        token: Discord bot token
+        ws: Websocket connected to Discord
+    """
+
+    def __init__(self, baseplate, token, actor_urn, *args, **kwargs):
+        """
+        Initialize DiscoBot
+
+        Args:
+            baseplate (Legobot.Lego): The parent Pykka actor.
+                Typically passed in from Legobot.Connectors.Discord.Discord
+            token (string): Discord bot token
+            actor_urn (string): URN of Pykka actor launching DiscoBot
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments.
+        """
+
+        self.baseplate = baseplate
+        self.rest_baseurl = 'https://discordapp.com/api/v6'
+        self.token = token
+        self.headers = {"Authorization": "Bot {}".format(token),
+                        "User-Agent": "Legobot",
+                        "Content-Type": "application/json"}
+        self.actor_urn = actor_urn
+        self.ws = None
+        threading.Thread.__init__(self)
+
+    def connect(self):
+        return create_connection(self.get_ws_url())
+
+    def create_message(self, channel_id, text):
+        """
+        Sends a message to a Discord channel or user via REST API
+
+        Args:
+            channel_id (string): ID of destingation Discord channel
+            text (string): Content of message
+        """
+
+        baseurl = self.rest_baseurl + \
+            '/channels/{}/messages'.format(channel_id)
+        requests.post(baseurl,
+                      headers=self.headers,
+                      data=json.dumps({'content': text}))
+
+    def identify(self, token):
+        """
+        Identifies to the websocket endpoint
+
+        Args:
+            token (string): Discord bot token
+        """
+
+        payload = {
```

Comment:
Again, I would comment this since the API doesn't make it clear what all of these parameters are doing.